src/main.rs

```diff
@@ -45,7 +46,40 @@ async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
     let url = Url::new(url, args.default_scheme);
     let host = url.host().unwrap();
     let method = method.unwrap_or(Method::from(&body)).into();
-    let auth = Auth::new(args.auth, args.auth_type, &host);
+    let mut auth = Auth::new(args.auth, args.auth_type, &host);
+
+    // load previous session if present
+    let arg_session = args.session.clone();
+    let previous_session = match args.session {
+        None => None,
+        Some(identifier) => {
+            match Session::load(&identifier, &host) {
+                Err(why) => panic!("couldn't load session {}: {}", &identifier, why),
+                Ok(result) => result,
+            }
+        },
+    };
+    let session_for_merge = previous_session.clone();
+    // Use auth from previous session if no auth present
+    match (auth.is_none(), previous_session) {
+        (true, Some(p)) => auth = p.auth,
+        (_, _) => (),
+    };
+    let saved_auth = auth.clone();
+    // Merge headers from parameters and previous session
+    let (headers, headers_to_unset) = request_items.headers(&session_for_merge);
+    // Save the current session if present
+    match arg_session {
+        None => (),
+        Some(identifier) => {
+            let new_session = Session::new(identifier, &host, request_items.export_headers(&session_for_merge), saved_auth);
+            match new_session.save() {
```

Comment:
This could be `if let Err(why) = new_session.save() { ... }` if removing the `Ok` arm (which doesn't have a meaningful body) is desired.

Reply:
Right, thanks again.